src/shared/components/VideoPlayer/VideoPlayer.tsx

```diff
@@ -115,37 +158,201 @@ const VideoPlayerComponent: React.ForwardRefRenderFunction<HTMLVideoElement, Vid
     player.play()
   }
 
+  useEffect(() => {
+    if (!player) {
+      return
+    }
+    const handler = () => setVideoTime(Math.floor(player.currentTime()))
+
+    player.on('timeupdate', handler)
+    return () => {
+      player.off('timeupdate', handler)
+    }
+  }, [player])
+
+  useEffect(() => {
+    if (!player) {
+      return
+    }
+    const handler = () => {
+      if (player.isFullscreen()) {
+        setIsFullScreen(true)
+      } else {
+        setIsFullScreen(false)
+      }
+    }
+    player.on('fullscreenchange', handler)
+    return () => {
+      player.off('fullscreenchange', handler)
+    }
+  }, [player])
+
+  useEffect(() => {
+    if (!player) {
+      return
+    }
+    const handler = () => {
+      if (player.isFullscreen()) {
+        setIsFullScreen(true)
+      } else {
+        setIsFullScreen(false)
+      }
+    }
+    player.on('fullscreenchange', handler)
+    return () => {
+      player.off('fullscreenchange', handler)
+    }
+  }, [player])
+
+  useEffect(() => {
+    if (!player) {
+      return
+    }
+    const enterPiPhandler = () => {
+      setIsPiPEnabled(true)
+    }
+    const exitPiPhandler = () => {
+      setIsPiPEnabled(false)
+    }
+
+    player.on('enterpictureinpicture', enterPiPhandler)
+    player.on('leavepictureinpicture', exitPiPhandler)
+    return () => {
+      player.off('enterpictureinpicture', enterPiPhandler)
+      player.off('leavepictureinpicture', exitPiPhandler)
+    }
+  }, [player])
+
   const debouncedVolumeChange = useRef(
     debounce((volume: number) => {
-      updatePlayerVolume(volume)
+      updateCachedPlayerVolume(volume)
     }, 500)
```

Comment:
Let's tweak this to 125
```suggestion
    }, 125)
```

Reply:
half a second is too much